Build failure: fish on darwin

### Nixpkgs version

- Unstable (25.11)

### Steps to reproduce

nix build .#fish

### Can Hydra reproduce this build failure?

Hydra doesn’t try to build the package.

### Link to Hydra build job

_No response_

### Relevant log output

```console
nix build .#fish
error: builder for '/nix/store/xbcj54iv1y948j6qdqn1bpbwk9cmq9zh-fish-4.2.1.drv' failed with exit code 1;
       last 25 log lines:
       >     tests/checks/math.fish
       >     tests/checks/signal.fish
       >     tests/checks/fish_update_completions.fish
       >     tests/checks/pipeline-pgroup.fish
       >     tests/checks/job-control-noninteractive.fish
       >     tests/checks/sigint.fish
       >     tests/checks/config-paths-standalone.fish
       >     tests/checks/disown-parent.fish
       >     tests/checks/help-completions.fish
       >     tests/checks/message-localization.fish
       >     tests/checks/message-localization-tier-is-declared.fish
       >     tests/checks/colon-delimited-var.fish
       >     tests/checks/sigint2.fish
       >     tests/checks/po-files-well-formed.fish
       >     tests/checks/noshebang.fish
       >     tests/checks/trap.fish
       >     tests/checks/wait.fish
       >     tests/checks/check-completions.fish
       >     tests/checks/jobs.fish
       >     tests/checks/sphinx-man.fish
       >     tests/checks/sphinx-html.fish
       >     tests/checks/check-all-fish-files.fish
       > FAILED: [code=1] CMakeFiles/fish_run_tests /nix/var/nix/builds/nix-build-fish-4.2.1.drv-0/b/source/build/CMakeFiles/fish_run_tests
       > cd /nix/var/nix/builds/nix-build-fish-4.2.1.drv-0/b/source && /nix/var/nix/builds/nix-build-fish-4.2.1.drv-0/b/source/tests/test_driver.py /nix/var/nix/builds/nix-build-fish-4.2.1.drv-0/b/source/build && env FISH_CMAKE_BINARY_DIR=/nix/var/nix/builds/nix-build-fish-4.2.1.drv-0/b/source/build PREFIX=/nix/store/szj33n7galqwf5xax0ipqiva8c3fs6hk-fish-4.2.1 DOCDIR=/nix/store/b3q9c9gkh3bbfqz4ffgxxvzycvffsccp-fish-4.2.1-doc/share/doc/fish DATADIR=/nix/store/szj33n7galqwf5xax0ipqiva8c3fs6hk-fish-4.2.1/share SYSCONFDIR=/nix/store/szj33n7galqwf5xax0ipqiva8c3fs6hk-fish-4.2.1/etc BINDIR=/nix/store/szj33n7galqwf5xax0ipqiva8c3fs6hk-fish-4.2.1/bin CARGO_TARGET_DIR=/nix/var/nix/builds/nix-build-fish-4.2.1.drv-0/b/source/build/cargo/build CARGO_BUILD_RUSTC=/nix/store/58qfkg2idgx439l4mwxrd0hdxvrpqk0r-rustc-wrapper-1.90.0/bin/rustc RUSTFLAGS=\  FISH_SPHINX=/nix/store/h0rpygqj9vjj0ngzq6j54bg0yd8gmkbq-python3.13-sphinx-8.2.3/bin/sphinx-build FISH_USE_PREBUILT_DOCS=FALSE /nix/store/yplr33zihm2n3xy5f9qxvg700yjgzyqx-cargo-1.90.0/bin/cargo test --no-default-features --workspace --target-dir /nix/var/nix/builds/nix-build-fish-4.2.1.drv-0/b/source/build/cargo/build/aarch64-apple-darwin --target aarch64-apple-darwin --lib
       > ninja: build stopped: subcommand failed.
       For full logs, run 'nix log /nix/store/xbcj54iv1y948j6qdqn1bpbwk9cmq9zh-fish-4.2.1.drv'.
```

### Additional context

git log|head
commit 5dc66d4a0db024ebeca7460995a7df34d96988b8
Merge: 836cae07645e 3cb6684582bf
Author: Gaétan Lepage <gaetan@glepage.com>
Date:   Fri Nov 14 00:08:36 2025 +0000

    COSMIC beta 6 (#461020)


### System metadata

 - system: `"aarch64-darwin"`
 - host os: `Darwin 24.6.0, macOS 15.7.3`
 - multi-user?: `yes`
 - sandbox: `no`
 - version: `nix-env (Lix, like Nix) 2.93.3
System type: aarch64-darwin
Additional system types: x86_64-darwin
Features: gc, signed-caches
System configuration file: /etc/nix/nix.conf
User configuration files: /Users/user/.config/nix/nix.conf:/Users/user/.nix-profile/etc/xdg/nix/nix.conf:/etc/profiles/per-user/user/etc/xdg/nix/nix.conf:/run/current-system/sw/etc/xdg/nix/nix.conf:/nix/var/nix/profiles/default/etc/xdg/nix/nix.conf
Store directory: /nix/store
State directory: /nix/var/nix
Data directory: /nix/store/wags9ajivrc1r24z9qx5glny5n8q5i4g-lix-2.93.3/share`
 - nixpkgs: `/nix/store/yff8cp4mrjrf71i1g9llxqpip8a6sfyj-source`


### Notify maintainers

@adamcstephens @cole-h @winterqt @SigmaSquadron @R-VdP

---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)


### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aopen+is%3Aissue+label%3A%220.kind%3A+build+failure%22). 
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/by-name/fi/fish/package.nix
+++ b/pkgs/by-name/fi/fish/package.nix
@@ -331,7 +331,8 @@ stdenv.mkDerivation (finalAttrs: {
   ]
   ++ lib.optional (!stdenv.hostPlatform.isDarwin) man-db;
 
-  doCheck = true;
+  # disable darwin pending https://github.com/NixOS/nixpkgs/pull/462090 getting through staging
+  doCheck = !stdenv.hostPlatform.isDarwin;
 
   nativeCheckInputs = [
     coreutils

